### PR TITLE
Patch for omnitrace-sample crash with HIP API.

### PR DESCRIPTION
**Pull request by ajanicijamd**

### CODE REVIEW

File: source/lib/omnitrace/library/runtime.cpp [hunk 1]
```diff
@@ -151,8 +151,19 @@ create_cpu_cid_entry(int64_t _tid)
     auto_lock_t _p_lk{ _p_mtx, std::defer_lock };
     if(!_p_lk.owns_lock()) _p_lk.lock();
 
-    auto&&     _cid        = get_cpu_cid()++;
-    auto&&     _parent_cid = get_cpu_cid_stack(_p_idx)->back();
+    auto&& _cid = get_cpu_cid()++;
+    // auto&&     _parent_cid = get_cpu_cid_stack(_p_idx)->back();
+    uint64_t _parent_cid;
```
Comment (jrmadsen) [inline source/lib/omnitrace/library/runtime.cpp:156]:
```suggestion
    uint64_t _parent_cid = 0;
```

**Review by jrmadsen — CHANGES_REQUESTED** [review]
LGTM except the variable initialization

**Review by jrmadsen — APPROVED** [review]

**Review by dgaliffiAMD — APPROVED** [review]

**Comment by ajanicijamd** [comment]
Removed the unnecessary else.
I performed a squash merge and enabled signing.

**Review by jrmadsen — APPROVED** [review]

**Review by dgaliffiAMD — APPROVED** [review]
Looks good. All verified now. I'm still not sure why there is that second merge commit, though.

**Comment by ajanicijamd** [comment]
I think that is because the main branch on my fork was behind the main branch on the omnitrace repo, so I had to merge this morning.

**Pull request merged**